### DISCUSSION TITLE
Provide an explicit tracking of (monetary) values through goal or event tracking

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#256] [Anything Else]
Allow the option to assign a monetary value to the conversion when setting up a goal or event. Similar to GA:

"Each time the goal is completed by a user, this amount is added together and seen in your reports as the Goal Value. For example, if your sales team can close 10% of people who sign up for a newsletter, and your average transaction is $500, you might assign $50 (i.e. 10% of $500) to your newsletter sign-up goal—a goal that users complete when they reach the final newsletter sign-up page. In contrast, if only 1% of signups result in a sale, you might only assign $5 to your newsletter sign-up goal."

(feature request via dm)

**Update:** We've now opened up a preview version of this to everyone. You can now assign monetary values to custom events and track the revenue attribution of your ecommerce store. Instructions at https://plausible.io/docs/ecommerce-revenue-tracking

**Comment by metmarkosaric** [comment]
Another request for this via email today: "the ability to assign a value to a goal and see the accumulated value for all goal completions". it's an important need for ecommerce.

**Reply by metmarkosaric** [reply]
Another more specific request: "Is it possible with Plausible to track the value in abandoned carts and the number of value in wishlists"

**Comment by metmarkosaric** [comment]
"Indeed like in Google Analytics. Assign a dynamic value to a goal (value is filled in via the dataLayer). Have the ability to sum up the values per dimension (per date, per source, per device, per country, ...)"

**Comment by gijo-varghese** [comment]
Would love to see this. We want to add the amount when someone purchases via the checkout.

**Comment by LauGau** [comment]
Yes, I think it's an important feature.

**Comment by ccgee** [comment]
Being able to **track event value for conversions is deal breaker** as far as many of my clients are concerned to switch over from GA

**Reply by rdvo** [reply]
yeah same here. I run lead-gen campaigns and being able to see how much revenue i'm getting per conversion would be awesome +1 please

**Comment by Aulig** [comment]
On a similar note: I use plausible to track my core web vitals and other numerical event values. It'd be useful to be able to calculate an average of those and maybe even graph it, as the current way of displaying it is not useful on it's own - I think I'll write a JS bookmarklet for now ;) :
![grafik](https://user-images.githubusercontent.com/32148066/179200677-423209c3-fe8d-4e3a-9147-df25623c3362.png)

**Reply by rdvo** [reply]
it's cool it can drill down by the revenue but no tally is tough..

**Comment by metmarkosaric** [comment]
**Update:** We've now opened up a preview version of this to everyone. You can now assign monetary values to custom events and track the revenue attribution of your ecommerce store. Instructions at https://plausible.io/docs/ecommerce-revenue-tracking

**Reply by rodrigotassinari** [reply]
hi @metmarkosaric great news that Plausible started supporting revenue attribution on events!

I've been testing it and have a couple of questions:

1. Events sent with revenue information ~~seem to be ignoring any custom properties sent with the event~~ don't allow breakdowns by properties. On the dashboard, the events which are set up with revenue tracking enabled don't show any possibility of breakdown by properties. However I can see and use the event's properties via the filter menu. Is that a bug, by design or just something you haven't been able to do yet?
2. There does not seem to be a "recalculation" of reporting amounts on past events when I change the reporting currency of an event. If I create a goal using BRL for reporting, than later delete that goal and create a new one with the same event name but a different reporting currency (say, USD for example), the dashboard shows the exact same amounts as before but with the new currency. I'd expect the system to re-convert the events' reporting amounts from the original currency to the new reporting currency (since the database stores both currencies and amounts after ingestion). Is this something you plan to support?

**Reply by dsager** [reply]
Ran into the same issue with 1. - It'd be quite useful to be able to break down further!
To give you a better idea, we already have a purchase event with a couple of properties.
![image](https://github.com/plausible/analytics/assets/1056502/90cdba42-dd91-4743-8e6f-332eaed733cc)
I now added the revenue payload to the event, but once I enable "revenue tracking" for this goal, all of these properties disappear:
![image](https://github.com/plausible/analytics/assets/1056502/1bd5c811-42d1-4478-9a74-e0bb5a0905c1)


In general, properties and filtering by them feel quite limited. For example, when filtering by a property, it'd be useful to see the breakdown of all the other properties, but it's not displayed. Like this and without the ability to filter by the combination of multiple properties, the feature has limited usefulness. Hoping that there are further iterations planned for this feature, especially before making this business tier feature with increased pricing!

Regarding point 2 - I guess it's converted at storage time, which makes sense: If you convert it a year later, the exchange rate might be completely different than it was at the time of recording it...

**Comment by metmarkosaric** [comment]
thanks @rodrigotassinari and @dsager! it's now possible to add more details to your revenue tracking by using custom properties. this lets you track order IDs, coupon codes, if customers are logged in and whatever else you need. our docs now include examples of how to do this with Shopify and WooCommerce. see https://plausible.io/docs/ecommerce-revenue-tracking

**Comment by metmarkosaric** [comment]
**Update:** Latest version of our WordPress plugin introduces a WooCommerce integration for revenue attribution tracking. The WooCommerce integration automatically sets up several custom events (including add to cart and remove from cart), revenue tracking for completed purchases, custom properties (including product name, price and total cart items) and a checkout funnel to help you understand the cart abandonment rate. No need to manually set up anything nor to edit your code. Details at: https://plausible.io/wordpress-analytics-plugin
